# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1258857258 18000
#      Sat Nov 21 21:34:18 2009 -0500
# Node ID 722bd8f6a750de5da34c1c46b6b507ea798dd3ba
# Parent  edf065b51fa94456d4328017a183e559ac725110
configure.ac: fix freetype check

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,14 @@
+2009-11-21  John W. Eaton  <jwe@octave.org>
+
+	* common.mk: Substitute FT2_CFLAGS.
+	* configure.ac: Remove spurious lines in check for freetype.  Add
+	$FT2_CFLAGS to $XTRA_CXXFLAGS.
+
 2009-11-18  Jaroslav Hajek  <highegg@gmail.com>
 
 	* NEWS: Update.
 
 2009-11-18  Jaroslav Hajek  <highegg@gmail.com>
 
 	* NEWS: Update.
 
diff --git a/common.mk b/common.mk
--- a/common.mk
+++ b/common.mk
@@ -161,16 +161,17 @@ RLD_FLAG = @RLD_FLAG@
 
 FLIBS = @FLIBS@
 
 LIBGLOB = @LIBGLOB@
 LIBOCTINTERP = @LIBOCTINTERP@
 LIBOCTAVE = @LIBOCTAVE@
 LIBCRUFT = @LIBCRUFT@
 
+FT2_CFLAGS = @FT2_CFLAGS@
 FT2_LIBS = @FT2_LIBS@
 
 HDF5_CPPFLAGS = @HDF5_CPPFLAGS@
 HDF5_LDFLAGS = @HDF5_LDFLAGS@
 HDF5_LIBS = @HDF5_LIBS@
 
 Z_CPPFLAGS = @Z_CPPFLAGS@
 Z_LDFLAGS = @Z_LDFLAGS@
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -741,39 +741,32 @@ AC_SUBST(MAGICK_LIBS)
 ## libraries needed for graphics
 
 warn_freetype=""
 
 OCTAVE_OPENGL
 if test -n "$OPENGL_LIBS"; then
   AC_DEFINE(HAVE_OPENGL, 1, [Define if OpenGL is available])
 
-## freetype 2
-  AC_CHECK_FT2([9.0.3],[AC_DEFINE(HAVE_FREETYPE, 1, [Define to 1 if you have Freetype library.])],
+  ## freetype 2
+
+  AC_CHECK_FT2([9.0.3], [AC_DEFINE(HAVE_FREETYPE, 1, [Define to 1 if you have Freetype library.])
+    XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"],
     [warn_freetype="FreeType library not found.  Native renderer will not have on-screen text"])
-  if test -z "$warn_freetype"; then
-    AC_LANG_PUSH(C++)
-      save_LIBS="$LIBS"
-      save_CXXFLAGS="$CXXFLAGS"
-      save_CPPFLAGS="$CPPFLAGS"
-      LIBS="$LIBS $FT2_LIBS $OPENGL_LIBS"
-      CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
-      CPPFLAGS="$CPPFLAGS $FT2_CFLAGS"
-  fi
 fi
 
-# fontconfig library
+## fontconfig library
 
 warn_fontconfig=""
 if test -z "$warn_freetype"; then
-  PKG_CHECK_MODULES(FONTCONFIG,[fontconfig],[
+  PKG_CHECK_MODULES(FONTCONFIG, [fontconfig], [
     have_fontconfig=yes
     OPENGL_LIBS="$FONTCONFIG_LIBS $OPENGL_LIBS"
     XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FONTCONFIG_CFLAGS"
-    AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present])],[
+    AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present])], [
     have_fontconfig=no
     warn_fontconfig="Fontconfig not found.  Native text rendering will use hard-coded font instead."])
 fi
 
 GRAPHICS_LIBS=
 GRAPHICS_CFLAGS=
 
 ## fltk (www.fltk.org)
